components/profile/unclaimed-hypercert-butchClaim-button.tsx

```diff
@@ -0,0 +1,145 @@
+    units: records.map((record) => BigInt(record.units!)),
+    proofs: records.map((record) => record.proof as (Hex | ByteArray)[]),
+    roots: records.map((record) => record.root as Hex | ByteArray),
+  };
+}
+
+export default function UnclaimedHypercertBatchClaimButton({
+  allowListRecords,
+  selectedChainId,
+}: {
+  allowListRecords: AllowListRecord[];
+  selectedChainId: number | null;
+}) {
+  const { client } = useHypercertClient();
+  const { data: walletClient } = useWalletClient();
+  const account = useAccount();
+  const { refresh } = useRouter();
+  const [isLoading, setIsLoading] = useState(false);
+  const { setDialogStep, setSteps, setOpen, setTitle } =
+    useStepProcessDialogContext();
+  const { switchChain } = useSwitchChain();
+
+  const selectedChain = selectedChainId
+    ? ChainFactory.getChain(selectedChainId)
+    : null;
+
+  const claimHypercert = async () => {
+    setIsLoading(true);
+    setOpen(true);
+    setSteps([
+      { id: "preparing", description: "Preparing to claim hypercert..." },
+      { id: "claiming", description: "Claiming hypercert on-chain..." },
+      { id: "confirming", description: "Waiting for on-chain confirmation" },
+      { id: "done", description: "Claiming complete!" },
+    ]);
+    setTitle("Claim Hypercert from Allowlist");
+    if (!client) {
+      throw new Error("No client found");
+    }
+    if (!walletClient) {
+      throw new Error("No wallet client found");
+    }
+    if (!account) {
+      throw new Error("No address found");
+    }
+
+    const claimData = transformAllowListRecords(allowListRecords);
+    await setDialogStep("preparing, active");
+    console.log(allowListRecords);
+    try {
+      await setDialogStep("claiming", "active");
+
+      const tx = await client.batchClaimFractionsFromAllowlists(claimData);
+      console.log(tx);
+      if (!tx) {
+        await setDialogStep("claiming", "error");
+        throw new Error("Failed to claim hypercert");
+      }
+      await setDialogStep("confirming", "active");
+      const receipt = await waitForTransactionReceipt(walletClient, {
+        confirmations: 3,
+        hash: tx,
+      });
+      if (receipt.status == "success") {
+        await setDialogStep("done", "completed");
+        await revalidatePathServerAction([
+          `/profile/${account.address}`,
+          `/profile/${account.address}?tab=hypercerts-claimable`,
+        ]);
+      } else if (receipt.status == "reverted") {
+        await setDialogStep("confirming", "error", "Transaction reverted");
+      }
+      console.log({ receipt });
+      setTimeout(() => {
+        refresh();
+      }, 5000);
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+  return (
+    <>
+      {account.chainId === selectedChainId ? (
+        <Button
+          variant={"outline"}
+          size={"sm"}
+          onClick={claimHypercert}
+          disabled={
```

Comment:
I personally prefer to take this kind of logic out of the JSX template and into the component function body, such as `const isClaimDisabled = isLoading ||`. Much easier to read later and keeps some distance between templating and logic.